encode/avcenc.c

```diff
@@ -928,6 +962,12 @@ int avcenc_render_picture()
     if (avcenc_context.misc_parameter_roi_buf_id != VA_INVALID_ID)
         va_buffers[num_va_buffers++] =  avcenc_context.misc_parameter_roi_buf_id;
 
+    if (avcenc_context.packed_aud_header_param_buf_id != VA_INVALID_ID)
+        va_buffers[num_va_buffers++] =  avcenc_context.packed_aud_header_param_buf_id;
+
+    if (avcenc_context.packed_aud_buf_id != VA_INVALID_ID)
+        va_buffers[num_va_buffers++] =  avcenc_context.packed_aud_buf_id;
+
```

Comment:
The order is the matter, the buffers are submitted by the following function call. BTW we can't assume the behavior of the driver.

       va_status = vaRenderPicture(va_dpy,
                                avcenc_context.context_id,
                                va_buffers,
                                num_va_buffers);